apps/secret/inline/v1beta1/inline.proto

```diff
@@ -0,0 +1,79 @@
+syntax = "proto3";
+
+package inlinev1beta1;
+
+option go_package = "github.com/grafana/grafana/apps/secret/inline/v1beta1;inlinev1beta1";
+
+message ObjectReference {
+    // APIGroup is the name of the API group that contains the referred object.
+    // The empty string represents the core API group.
+    string api_group = 1;
+
+    // APIVersion is the version of the API group that contains the referred object.
+    string api_version = 2;
+
+    // See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+    string kind = 3;
+
+    // Tenant isolation
+    string namespace = 4;
+
+    // Explicit resource identifier
+    string name = 5;
+}
+
+message InlineSecureValue {
+    // Create a secure value -- this is only used for POST/PUT
+    optional string create = 1;
+
+    // Name in the secret service (reference)
+    optional string name = 2;
+
+    // Remove this value from the secure value map
+    // Values owned by this resource will be deleted if necessary
+    optional bool remove = 3;
+}
+
+message CanReferenceRequest {
+    // Owner reference.
+    ObjectReference owner = 1;
+ 
+    // Key is the field in the owner resource.
+    map<string, InlineSecureValue> values = 2; 
```

Comment:
IIUC, this could be repeated string.  The key create, and delete are invalid inputs 🤔